service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java

```diff
@@ -17,28 +20,42 @@
 @Service
 public class CommandManager {
     private final TerraformManager terraformManager;
-
+    private final CommandMapper commandMapper;
+    private final CommandPublisher commandPublisher;
     private final MessagingServiceDelegateService messagingServiceDelegateService;
+    private final EventPortalProperties eventPortalProperties;
 
-    public CommandManager(TerraformManager terraformManager, MessagingServiceDelegateService messagingServiceDelegateService) {
+    public CommandManager(TerraformManager terraformManager, CommandMapper commandMapper,
+                          CommandPublisher commandPublisher, MessagingServiceDelegateService messagingServiceDelegateService,
+                          EventPortalProperties eventPortalProperties) {
         this.terraformManager = terraformManager;
+        this.commandMapper = commandMapper;
+        this.commandPublisher = commandPublisher;
         this.messagingServiceDelegateService = messagingServiceDelegateService;
+        this.eventPortalProperties = eventPortalProperties;
     }
 
-    public void execute(CommandRequest request) {
+    public void execute(CommandMessage request) {
         Map<String, String> envVars = setBrokerSpecificEnvVars(request.getServiceId());
         for (CommandBundle bundle : request.getCommandBundles()) {
             // For now everything is run serially
             for (Command command : bundle.getCommands()) {
                 switch (command.getCommandType()) {
                     case terraform:
-                        terraformManager.execute(request, command, envVars);
+                        terraformManager.execute(commandMapper.map(request), command, envVars);
```

Comment:
Map the CommandMessage to our internal CommandRequest type so the terraform plugin doesn't need to know about mop messages.